pkg/kinflate/tree/builder.go

```diff
@@ -0,0 +1,200 @@
+	if err != nil {
+		return nil, err
+	}
+	m, err := manifestPathToManifest(path)
+	if err != nil {
+		return nil, err
+	}
+	return manifestToManifestNode(m)
+}
+
+func manifestToManifestNode(m *manifest.Manifest) (*ManifestNode, error) {
+	mnode := &ManifestNode{}
+	var err error
+	mnode.Data, err = manifestToManifestData(m)
+	if err != nil {
+		return nil, err
+	}
+
+	mnode.Children = []*ManifestNode{}
+	for _, pkg := range m.Packages {
+		child, err := manifestPathToManifestNode(pkg)
+		if err != nil {
+			return nil, err
+		}
+		mnode.Children = append(mnode.Children, child)
+	}
+	return mnode, nil
+}
+
+// manifestPathToManifest loads a manifest file and parse it in to the Manifest object.
+func manifestPathToManifest(filename string) (*manifest.Manifest, error) {
+	bytes, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	var m manifest.Manifest
+	// TODO: support json
+	err = yaml.Unmarshal(bytes, &m)
+	if err != nil {
+		return nil, err
+	}
+	dir, _ := path.Split(filename)
+	adjustpath.AdjustPathsForManifest(&m, []string{dir})
+	return &m, err
+}
+
+// validateManifestPath loads the manifest from the given path.
+// It returns ManifestData and an potential error.
+func validateManifestPath(mPath string) (string, error) {
+	f, err := os.Stat(mPath)
+	if err != nil {
+		return "", err
+	}
+	if f.IsDir() {
+		mPath = path.Join(mPath, constants.KubeManifestFileName)
+		_, err = os.Stat(mPath)
+		if err != nil {
+			return "", err
+		}
+	} else {
+		if !strings.HasSuffix(mPath, constants.KubeManifestFileName) {
+			return "", fmt.Errorf("expecting file: %q, but got: %q", constants.KubeManifestFileName, mPath)
+		}
+	}
+	return mPath, nil
+}
+
+func manifestToManifestData(m *manifest.Manifest) (*ManifestData, error) {
+	mdata := &ManifestData{}
+	var err error
+	mdata.Name = m.Name
+	mdata.NamePrefix = NamePrefixType(m.NamePrefix)
+	mdata.ObjectLabels = m.ObjectLabels
+	mdata.ObjectAnnotations = m.ObjectAnnotations
+	mdata.Resources, err = pathsToMap(m.Resources)
+	if err != nil {
+		return nil, err
+	}
+	mdata.Patches, err = pathsToMap(m.Patches)
+	if err != nil {
+		return nil, err
+	}
+	mdata.Configmaps, err = cutil.MakeMapOfConfigMap(m)
+	if err != nil {
+		return nil, err
+	}
+	mdata.Secrets, err = cutil.MakeMapOfGenericSecret(m)
+	if err != nil {
+		return nil, err
+	}
+	TLSSecrets, err := cutil.MakeMapOfTLSSecret(m)
+	err = mergemap.Merge(mdata.Secrets, TLSSecrets)
+	if err != nil {
+		return nil, err
+	}
+	return mdata, nil
+}
+
+func pathsToMap(paths []string) (map[gvkn.GroupVersionKindName]*unstructured.Unstructured, error) {
+	res := map[gvkn.GroupVersionKindName]*unstructured.Unstructured{}
+	for _, path := range paths {
+		err := pathToMap(path, res)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return res, nil
+}
+
+func pathToMap(path string, into map[gvkn.GroupVersionKindName]*unstructured.Unstructured) error {
+	_, err := os.Stat(path)
+	if err != nil {
+		return err
+	}
+	if into == nil {
```

Comment:
This should be an error.

Alternatively, this method doesn't have to take KObject as an arg.  It could just create one, and the caller could merge it into some other KObject.